concrete/src/Http/Middleware/SecurityHeaderMiddleware.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Concrete\Core\Http\Middleware;
+
+use Concrete\Core\Config\Repository\Repository;
+use Concrete\Core\Utility\Service\Validation\Strings;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Middleware for applying security related headers
+ * @package Concrete\Core\Http\Middleware
+ */
+class SecurityHeaderMiddleware implements MiddlewareInterface
+{
+
+    /**
+     * @var Repository
+     */
+    private $config;
+
+    /**
+     * @var Strings
+     */
+    private $stringValidator;
+
+    public function __construct(Repository $config, Strings $stringValidator)
+    {
+        $this->config = $config;
+        $this->stringValidator = $stringValidator;
+    }
+
+    /**
+     * @param DelegateInterface $frame
+     * @return Response
+     */
+    public function process(Request $request, DelegateInterface $frame)
+    {
+        $response = $frame->next($request);
+        $options = $this->config->get('concrete.security.misc');
+
+        if ($response->headers->has('Content-Security-Policy') === false) {
+            $content_security_policies = $options['content_security_policy'];
```

Comment:
This would throw a notice if `content_security_policy` isn't set for any reason. Use the null coalesce operator to prevent this: `$options['content_security_policy'] ?? false`